---
title: "Figures_Bowtie"
author: "marie"
date: "6 dec 2018"
output: html_document
---

```{r setup, include=FALSE, warning =F, message=F}
knitr::opts_chunk$set(echo = TRUE, warning =F, message=F)
library(Rmisc)
library(ggplot2)
library(cowplot)
 library(reshape2)
library(dplyr)
library("DESeq2")


```

```{r input, include=F, warning =F, message=F}

###### Installations #######################################################################################################################
setwd("~/Documents/StageMaurine/QuantifBowtie/")



###### Importation des donnees sorties de Bowtie #############################################################################################
samples <- read.csv("~/Documents/StageMaurine/samples.txt", header = TRUE, sep = "\t", row.names=1) #conditions de chaque echantillon
#Type = souche
#Row = mx ou md
#Weight = stade developpemental

MusFVBId <- read.csv("~/Documents/StageMaurine/corresp_FVB",header=F,sep="|") #EnsembleID_FVB et MGI
MusFVBId <- MusFVBId[,c(2,3)]
names(MusFVBId)=c("Transcript.stable.ID", "MGI.symbol")

MusId <- read.csv("~/Documents/StageMaurine/Biomart_GRCm38.p5.txt",header=TRUE,sep="\t") #EnsembleID et MGI
filesMus <- Sys.glob("bowtie/SHPC*.idxstats.txt") 

## Creation tx2gene
tx2gene <- MusId[,c("Transcript.stable.ID", "MGI.symbol")]
tx2Spike <- read.csv("~/Documents/StageMaurine/CaracteristicsERCC.txt", header=FALSE, sep="\t")
tx2Spike <- tx2Spike[-1,-3]
tx2Spike$V2 <- tx2Spike$V1
colnames(tx2Spike) <- colnames(tx2gene) 
tx2gene <- rbind(tx2gene,tx2Spike) #Ajout des noms des spike au tx2gene
tx2geneFVB <- rbind(MusFVBId,tx2Spike) #Ajout des noms des spike au tx2gene



id=gsub("bowtie/","",filesMus)
id=gsub(".idxstats.txt","",id)

lFVB=read.csv(filesMus[grep("FVB",id)][1], header = FALSE, sep = "\t")[,1:2]
lFVB1=lapply(filesMus[grep("FVB",id)],function(x){t=read.csv(x, header = FALSE, sep = "\t");t[,3]})
dFVB=cbind(lFVB,data.frame(lFVB1))
names(dFVB)=c("Transcript","length",id[grep("FVB",id)])
dFVB1=merge(tx2geneFVB,dFVB,by.x="Transcript.stable.ID",by.y="Transcript")
dFVB1= dFVB1 %>% group_by(MGI.symbol) %>% summarise_if(is.numeric,sum)

lMUS=read.csv(filesMus[-grep("FVB",id)][1], header = FALSE, sep = "\t")[,1:2]
lMUS1=lapply(filesMus[-grep("FVB",id)],function(x){t=read.csv(x, header = FALSE, sep = "\t");t[,3]})
dMUS=cbind(lMUS,data.frame(lMUS1))
names(dMUS)=c("Transcript","length",id[-grep("FVB",id)])
dMUS$Transcript=gsub("(.*)(\\..*)","\\1",dMUS$Transcript)
dMUS1=merge(tx2gene,dMUS,by.x="Transcript.stable.ID",by.y="Transcript")
dMUS1= dMUS1 %>% group_by(MGI.symbol) %>% summarise_if(is.numeric,sum)



dtot=merge(dFVB1,dMUS1,by.x="MGI.symbol",by.y="MGI.symbol")
row.names(dtot)=dtot$MGI.symbol
dtot=dtot[,-1]
dtot=dtot[-1,]

plot(dtot$length.x,dtot$length.y,xlab="length annot ref genome",ylab="length annot FVB")

dMUS1=data.frame(dMUS1)
dFVB1=data.frame(dFVB1)
row.names(dMUS1)=dMUS1$MGI.symbol
row.names(dFVB1)=dFVB1$MGI.symbol
dMUS1=dMUS1[-1,-c(1:2)]
dFVB1=dFVB1[-1,-c(1:2)]
```


# Analyses spike-in

Analyses préliminaires avec uniquement le nombre de reads total qui s'aligne sur des spike-in, et barplot des comptes normalisés de chaque spike-in : on voit que les distributions n'ont pas la meme tête entre échantillons...

```{r spike,  warning =F, out.width = '100%'}
samplestot=data.frame(rbind(samples,samples))
samplestot$mapping=c(rep("FVB",12),rep("genome",12))
row.names(samplestot)=names(dtot[,-grep("length",names(dtot))])
ddstot <- DESeqDataSetFromMatrix(dtot[,-grep("length",names(dtot))], colData = samplestot, design = ~ 1)
ddstot <- DESeq(ddstot)


ct_long=melt(counts(ddstot,norm=T), id.vars=c("Row.names"))

ct_long_ercc=ct_long %>%  
    filter(grepl("ERCC",Var1))
  
p=ggplot(ct_long_ercc, aes(x=Var2, y=value, group=Var2)) +
geom_bar(stat="identity")+coord_flip()+ ylab("Sum Reads ERCC") + xlab("Sample")+
  theme_minimal() 
p


ct_long_ercc_classique=ct_long %>%  
    filter(grepl("ERCC",Var1)&!grepl("FVB",Var2))
  
p=ggplot(ct_long_ercc_classique, aes(x=Var1, y=value, group=Var2)) +
geom_bar(stat="identity")+coord_flip()+ ylab("Sum Reads ERCC") + xlab("Sample")+
  theme_minimal() + facet_wrap(~ Var2, ncol=2) +theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())
p


ct_long_ercc_classique_part=ct_long %>%  
    filter(grepl("ERCC-0001",Var1)&!grepl("FVB",Var2))
  
p=ggplot(ct_long_ercc_classique_part, aes(x=Var1, y=value, group=Var2)) +coord_flip()+
geom_bar(stat="identity")+ ylab("normalized level ERCC") + xlab("Sample")+
  theme_minimal() + facet_wrap(~ Var2, ncol=2) +theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())
p

```

# DESeq mapping artifact


- virer les gènes signif dans comptage FVB versus comptage C57BL6 -> me donner la table : 

Attention les différents jeux de données n'ont pas le meme nombre de gènes (MUS1 = référence genome, FVB = FVB, tot = merge des deux). Les résultats du test pour dtot sont dans results_mappingartifact_allgenes.txt, la liste des gènes signif est dans liste_mappingartifact.txt


```{r mapping, warning =F}

knitr::kable(dim(dMUS1))
knitr::kable(dim(dFVB1))
knitr::kable(dim(dtot))


ddstot <- DESeqDataSetFromMatrix(dtot[,-grep("length",names(dtot))], colData = samplestot, design = ~ mapping)
ddstot <- DESeq(ddstot)

resultsMapping <- results(ddstot)
knitr::kable(table(resultsMapping$padj < 0.05))
knitr::kable(table(resultsMapping$padj < 0.05,abs(dtot$length.x-dtot$length.y)>0.1*dtot$length.x))

c=counts(ddstot,norm=T)
cm=merge(resultsMapping,c,by.x=0,by.y=0)
row.names(cm)=row.names(resultsMapping)
write.table(file="results_mappingartifact_allgenes.txt",cm,row.names=T,sep="\t",quote=F)
write.table(file="liste_mappingartifact.txt",row.names(resultsMapping[resultsMapping$padj < 0.05&!is.na(resultsMapping$padj),]),row.names=F,col.names=F,sep="\t",quote=F)


write.table(file="liste_mappingartifact_length10per.txt",row.names(resultsMapping[resultsMapping$padj < 0.05&!is.na(resultsMapping$padj)&abs(dtot$length.x-dtot$length.y)>0.1*dtot$length.x,]),row.names=F,col.names=F,sep="\t",quote=F)


```

# Effet souche

Puis: sur gènes restants

## Sur Md+Mx

- gènes DE design Strain, prenant en compte la mâchoire -> me donner la table, si possible avec non seulement les gènes, la p-value, le fold change mais les comptes dans chaque échantillon



```{r jaw + strain, warning =F}

listemappingartifact=row.names(resultsMapping[resultsMapping$padj < 0.05&!is.na(resultsMapping$padj)&abs(dtot$length.x-dtot$length.y)>0.1*dtot$length.x,])

ddsMus <- DESeqDataSetFromMatrix(dMUS1[(!row.names(dMUS1)%in%listemappingartifact),], colData = samples, design = ~ Jaw+Type)
ddsMus <- DESeq(ddsMus)
resultsMus <- results(ddsMus)
knitr::kable(table(resultsMus$padj < 0.05))

c=counts(ddsMus,norm=T)
cm=merge(resultsMus,c,by.x=0,by.y=0)
write.table(file="results_MusknowingStrain_allgenes.txt",cm,row.names=F,sep="\t",quote=F)
write.table(file="liste_MusknowingStrain.txt",row.names(resultsMus[resultsMus$padj < 0.05&!is.na(resultsMus$padj),]),row.names=F,col.names=F,sep="\t",quote=F)

```


## Sur Mx

- gènes DE Strain dans mx -> me donner la table, si possible avec non seulement les gènes, la p-value, le fold change mais les comptes dans chaque échantillon


```{r Mx: strain, warning =F}


ddsMusMx <- DESeqDataSetFromMatrix(dMUS1[(!row.names(dMUS1)%in%listemappingartifact),samples$Jaw=="mX"], colData = samples[samples$Jaw=="mX",], design = ~ Type)
ddsMusMx <- DESeq(ddsMusMx)
resultsMusMx <- results(ddsMusMx)
knitr::kable(table(resultsMusMx$padj < 0.05))

c=counts(ddsMusMx,norm=T)
cm=merge(resultsMusMx,c,by.x=0,by.y=0)
write.table(file="results_MusMx_allgenes.txt",cm,row.names=T,sep="\t",quote=F)
write.table(file="liste_MusMx.txt",row.names(resultsMusMx[resultsMusMx$padj < 0.05&!is.na(resultsMusMx$padj),]),row.names=F,col.names=F,sep="\t",quote=F)

```

## Sur Md

- gènes DE Strain dans md -> me donner la table, si possible avec non seulement les gènes, la p-value, le fold change mais les comptes dans chaque échantillon



```{r Md: strain, warning =F}


ddsMusMd <- DESeqDataSetFromMatrix(dMUS1[(!row.names(dMUS1)%in%listemappingartifact),samples$Jaw=="mD"], colData = samples[samples$Jaw=="mD",], design = ~ Type)
ddsMusMd <- DESeq(ddsMusMd)
resultsMusMd <- results(ddsMusMd)
knitr::kable(table(resultsMusMd$padj < 0.05))

c=counts(ddsMusMd,norm=T)
cm=merge(resultsMusMd,c,by.x=0,by.y=0)
write.table(file="results_MusMd_allgenes.txt",cm,row.names=T,sep="\t",quote=F)
write.table(file="liste_MusMd.txt",row.names(resultsMusMx[resultsMusMd$padj < 0.05&!is.na(resultsMusMd$padj),]),row.names=F,col.names=F,sep="\t",quote=F)

```

